tests/unit/test_charm.py

```diff
@@ -39,13 +39,14 @@ def test_init_postgresql_relation(self):
         # We'll only test the case where _stored already
         # has content. _stored being empty is basically tested
         # by all the other functions
-
-        with patch('test_charm.GunicornK8sCharm._stored') as mock_stored:
-            with patch('pgsql.PostgreSQLClient'):
-                with patch('test_charm.GunicornK8sCharm.on'):
-                    mock_stored.reldata = {'pg': 'foo'}
-                    c = GunicornK8sCharm(MagicMock())
-                    self.assertEqual(c._stored.reldata, mock_stored.reldata)
+        with patch("charm.GunicornK8sCharm.__init__") as mock_init:
```

Comment:
`with` statements support multiple context managers: https://docs.python.org/3.1/reference/compound_stmts.html#with, do you think that would make it easier to read?